Dockerfile

```diff
@@ -1,21 +1,21 @@
 FROM golang:alpine as builder
 
-ADD . /usr/src/rdma-cni
+COPY . /usr/src/rdma-cni
 
 ENV HTTP_PROXY $http_proxy
 ENV HTTPS_PROXY $https_proxy
 
-RUN apk add --update --virtual build-dependencies build-base && \
-    cd /usr/src/rdma-cni && \
+WORKDIR /usr/src/rdma-cni
+RUN apk add --no-cache --virtual build-dependencies build-base=~0.5 && \
     make clean && \
     make build
 
-FROM alpine
+FROM alpine:3
```

Comment:
why in the build image you don't use https://hub.docker.com/layers/library/golang/1.20.4-alpine3.18/images/sha256-0a03b591c358a0bb02e39b93c30e955358dadd18dc507087a3b7f3912c17fe13?context=explore

Reply:
essentially copied either from sriov-cni or sriov-device-plugin.

in build image it will build with latest golang compiler.
we didnt hit any issues with it until now as it is.

Reply:
I think it better to use the same image from compile and run. 
so with should use explicitly golang:alpine3 and not golang:alpine
Once alpine:4 will be release I think this will break.

Reply:
there is no alpine3 tag.

currently it will build against latest version of go (which is generally good)  and will run on latest  version of alpine3 (technically what runs is just the shell script, CNI is invoked from host ).
which i think it fine. in case it breaks for some glibc when alpine4 is out we can handle it then.

LMK if current setting is fine or take your pick from : https://hub.docker.com/_/golang

Reply:
let keep it as is.